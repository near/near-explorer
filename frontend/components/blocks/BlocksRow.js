import Link from "next/link";

import { Row, Col } from "react-bootstrap";

import { DataConsumer } from "../utils/DataProvider";
import Timer from "../utils/Timer";

const BlocksRow = props => (
  <Link href="/blocks/[hash]" as={`/blocks/${props.block.hash}`}>
    <a style={{ textDecoration: "none" }}>
      <Row
        className={`transaction-row mx-0 ${
          props.cls === undefined ? "" : props.cls
        }`}
      >
        <Col md="auto" xs="1" className="pr-0">
          <img src="/static/images/icon-m-block.svg" />
        </Col>
        <Col md="auto" xs="1" className="pr-0">
          <img
            src="/static/images/icon-m-block.svg"
            style={{ width: "20px" }}
          />
        </Col>
        <Col md="7" xs="6">
          <Row>
            <Col className="transaction-row-title">#{props.block.height}</Col>
          </Row>
          <Row>
            <Col className="transaction-row-text">
              <Row>
                <Col md="auto">
                  <img
                    src="/static/images/icon-m-transaction.svg"
                    style={{ width: "12px" }}
                  />
                  {` ${props.block.transactionsCount}`}
                </Col>
              </Row>
            </Col>
          </Row>
        </Col>
        <Col md="3" xs="4" className="ml-auto text-right">
          <Row>
            <Col className="transaction-row-txid">
              {props.block.hash !== undefined
                ? `${props.block.hash.substring(0, 7)}...`
                : null}
            </Col>
          </Row>
          <Row>
            <Col className="transaction-row-timer">
              <span className="transaction-row-timer-status">Completed</span>
              &nbsp;&nbsp;
              <Timer time={props.block.timestamp} />
            </Col>
          </Row>
        </Col>
        <style jsx global>{`
          .transaction-row {
            padding-top: 10px;
            padding-bottom: 10px;
            border-top: solid 2px #f8f8f8;
          }

          .transaction-row:hover {
            background: rgba(0, 0, 0, 0.1);
          }

          .transaction-row-bottom {
            border-bottom: solid 2px #f8f8f8;
          }

          .transaction-row-title {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #24272a;
          }

          .transaction-row-text {
            font-family: BentonSans;
            font-size: 12px;
            font-weight: 500;
            line-height: 1.5;
            color: #999999;
          }

          .transaction-row-txid {
            font-family: BentonSans;
            font-size: 14px;
            font-weight: 500;
            line-height: 1.29;
            color: #0072ce;
          }

          .transaction-row-timer {
            font-family: BentonSans;
            font-size: 12px;
            color: #999999;
            font-weight: 100;
          }

          .transaction-row-timer-status {
            font-weight: 500;
          }
        `}</style>
      </Row>
    </a>
  </Link>
);

export default BlocksRow;
