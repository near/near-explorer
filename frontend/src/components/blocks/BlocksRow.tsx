import Link from "next/link";

import React from "react";

import { Row, Col } from "react-bootstrap";

import * as B from "../../libraries/explorer-wamp/blocks";

import Timer from "../utils/Timer";

export interface Props {
  block: B.BlockInfo;
  cls?: string;
}

export default class extends React.PureComponent<Props> {
  render() {
    const { block } = this.props;
    return (
      <Link href="/blocks/[hash]" as={`/blocks/${block.hash}`}>
        <a style={{ textDecoration: "none" }}>
          <Row
            className={`transaction-row mx-0 ${
              this.props.cls === undefined ? "" : this.props.cls
            }`}
          >
            <Col md="auto" xs="1" className="pr-0">
              <img src="/static/images/icon-m-block.svg" />
            </Col>
            <Col md="auto" xs="1" className="pr-0">
              <img
                src="/static/images/icon-m-block.svg"
                style={{ width: "20px" }}
              />
            </Col>
            <Col md="7" xs="6">
              <Row>
                <Col className="transaction-row-title">#{block.height}</Col>
              </Row>
              <Row>
                <Col className="transaction-row-text">
                  <Row>
                    <Col md="auto">
                      <img
                        src="/static/images/icon-m-transaction.svg"
                        style={{ width: "12px" }}
                      />
                      {` ${block.transactionsCount}`}
                    </Col>
                  </Row>
                </Col>
              </Row>
            </Col>
            <Col md="3" xs="4" className="ml-auto text-right">
              <Row>
                <Col className="transaction-row-txid">
                  {block.hash !== undefined
                    ? `${block.hash.substring(0, 7)}...`
                    : null}
                </Col>
              </Row>
              <Row>
                <Col className="transaction-row-timer">
                  <span className="transaction-row-timer-status">
                    Completed
                  </span>
                  &nbsp;&nbsp;
                  <Timer time={block.timestamp} />
                </Col>
              </Row>
            </Col>
            <style jsx global>{`
              .transaction-row {
                padding-top: 10px;
                padding-bottom: 10px;
                border-top: solid 2px #f8f8f8;
              }

              .transaction-row:hover {
                background: rgba(0, 0, 0, 0.1);
              }

              .transaction-row-title {
                font-family: BentonSans;
                font-size: 14px;
                font-weight: 500;
                line-height: 1.29;
                color: #24272a;
              }

              .transaction-row-text {
                font-family: BentonSans;
                font-size: 12px;
                font-weight: 500;
                line-height: 1.5;
                color: #999999;
              }

              .transaction-row-txid {
                font-family: BentonSans;
                font-size: 14px;
                font-weight: 500;
                line-height: 1.29;
                color: #0072ce;
              }

              .transaction-row-timer {
                font-family: BentonSans;
                font-size: 12px;
                color: #999999;
                font-weight: 100;
              }

              .transaction-row-timer-status {
                font-weight: 500;
              }
            `}</style>
          </Row>
        </a>
      </Link>
    );
  }
}
