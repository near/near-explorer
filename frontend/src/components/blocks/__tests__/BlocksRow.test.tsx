import renderer from "react-test-renderer";

import BlocksRow from "../BlocksRow";

describe("<BlocksRow />", () => {
  it("renders", () => {
    expect(
      renderer.create(
        <BlocksRow
          block={{
            transactionsCount: 3,
            timestamp: Number(new Date(2019, 1, 1)),
            hash: "EVvWW1S9BFaEjY1JBNSdstb7ZTtTFjQ6cygkbw1KY4tL",
            prevHash: "EVvWW1S9BFaEjY1JBNSdstb7Zjghjlyguiygfhgu",
            height: 12345,
            gasPrice: "5000",
            gasUsed: 0,
            isFinal: false,
          }}
        />
      )
    ).toMatchSnapshot();
  });
});
