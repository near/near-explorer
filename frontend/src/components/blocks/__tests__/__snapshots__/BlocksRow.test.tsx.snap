// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlocksRow /> renders 1`] = `
<a
  href="/blocks/EVvWW1S9BFaEjY1JBNSdstb7ZTtTFjQ6cygkbw1KY4tL"
  onClick={[Function]}
  onMouseEnter={[Function]}
  style={
    Object {
      "textDecoration": "none",
    }
  }
>
  <div
    className="transaction-row mx-0  row"
  >
    <div
      className="pr-0 col-md-auto col-1"
    >
      <img
        className="jsx-324075529"
        src="/static/images/icon-m-block.svg"
      />
    </div>
    <div
      className="pr-0 col-md-auto col-1"
    >
      <img
        className="jsx-324075529"
        src="/static/images/icon-m-block.svg"
        style={
          Object {
            "width": "20px",
          }
        }
      />
    </div>
    <div
      className="col-md-7 col-6"
    >
      <div
        className="row"
      >
        <div
          className="transaction-row-title col"
        >
          #
          12345
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="transaction-row-text col"
        >
          <div
            className="row"
          >
            <div
              className="col-md-auto"
            >
              <img
                className="jsx-324075529"
                src="/static/images/icon-m-transaction.svg"
                style={
                  Object {
                    "width": "12px",
                  }
                }
              />
               3
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="ml-auto text-right col-md-3 col-4"
    >
      <div
        className="row"
      >
        <div
          className="transaction-row-txid col"
        >
          EVvWW1S...
        </div>
      </div>
      <div
        className="row"
      >
        <div
          className="transaction-row-timer col"
        >
          <span
            className="jsx-324075529 transaction-row-timer-status"
          >
            Finalizing
          </span>
            
          <span>
            1s ago
          </span>
        </div>
      </div>
    </div>
  </div>
</a>
`;
